### ISSUE TITLE
Organize repo

### ISSUE THREAD

**Issue by sgosline** [Sage-Bionetworks/synapseAnnotations#71]
We should organize the repos

- create directory structure for binaries and schema
- make sure there are jsons and remove ymls
- Update readme to describe how to use these schemas.

**Comment by jaeddy** [comment]
I'm inclined to organize the repo following standard structure for a Python package. This might be overkill, as we might not ever need to import or work with annotation "objects" within Python, but this would leave us with the option to move in that direction.

The most relevant example is the GA4GH schemas repo/library:
https://github.com/ga4gh/ga4gh-schemas/tree/master/python

The purpose and use cases of their schemas are definitely more advanced and ambitious than ours will (likely) be; however, the idea of defining schemas in terms of Python classes would have some advantages in terms of testing, inheritance, and extensibility.

Here's my proposal for the initial re-org, with some comments below:

```
synapseAnnotations/
├── LICENSE
├── README.md
├── bin
├── docs
│   └── use_cases.yml
├── scripts
│   ├── createTermList.py
│   └── json2schema.py
├── synapseAnnotations
│   ├── __init__.py
│   └── data
│       ├── Neuro_Annotations.json
│       ├── genie_annotations.json
│       ├── minimal_Sage_analysis.yml
│       ├── minimal_Sage_standard.json
│       ├── minimal_Sage_standard.yml
│       ├── nf_annnotations.yml
│       └── nf_annotations.json
└── tests
```

- `synapseAnnotations` is where the main library code would go; because this code doesn't really exist yet, and the schemas currently only exist as files, I've placed them in a `data` subfolder and dropped in an empty `__init__.py` file. The library itself will likely comprise core functionality for working with schemas and schema files and (eventually, maybe) code/modules that actually define the schema (replacing the current JSON/YAML versions).  
- `bin` is where production-level wrapper scripts would live; these import the `synapseAnnotations` library (or individual modules) and perform common tasks/operations.
- `scripts` is a place for in-development code which might eventually become modules in the main library or production-level scripts in `bin`
- `docs` is obvious; I think because there will be a lot of use cases we want to document and demonstrate, building out a set of pages with Sphinx and ReadTheDocs could be nice.
- `tests` a friendly reminder that we should have them, in some form, sooner than later!

Not currently shown/included: other Python packaging elements (e.g., `setup.py`, etc.).

I think that's it for now. Please comment below.

**Comment by teslajoy** [comment]
^ very nice!

**Comment by kdaily** [comment]
![great success](https://media.giphy.com/media/a0h7sAqON67nO/giphy.gif)

**Comment by sgosline** [comment]
What would go in the scripts? Wouldn't all scripts go into synAnnotationUtils? Or should all those scripts come in here?

**Comment by jaeddy** [comment]
This probably gets a bit subjective, but the Python files in `scripts` or `bin` are designed to be called directly (from the command line), whereas `.py` files in the `synapseAnnotation` folder are *modules* containing functions or classes that can be accessed by importing the module. For example, if `json2schema.py` was a module, you would access the functions within Python using `from synapseAnnotations import json2schema`. Some modules can be multi-purpose: import-able or able to run directly as a script (i.e., `if __name__ == "__main__":`). A common workflow might be to test some new functionality by writing a script, then eventually convert it to a package module when it's working.

tl;dr: code in `synapseAnnotations` (the main package/application code) is designed to be accessed by other Python modules/scripts; scripts, on the other hand, provide a wrapper to access and use package code from the command line

**Comment by kdaily** [comment]
I think @sgosline is asking if we should keep one repo instead of this one and https://github.com/Sage-Bionetworks/synAnnotationUtils (which, incidentally has the necessary Python package setup including some scripts which are in the `bin` directory). I don't see the need for two repos.

**Comment by jaeddy** [comment]
Oh, right! Sorry @sgosline for not reading more carefully. Yes - I saw the `synAnnotationUtils` repo and wasn't sure what the distinction was between the two... We could certainly consider merging.

**Comment by teslajoy** [comment]
imagine a <Dr. Bronners all one hemp lavender soap>.png  👍

**Comment by sgosline** [comment]
Verdict: table the discussion about merging until next week, but re-organize the repo in the way proposed above.

**Comment by sgosline** [comment]
I think we've decided to avoid hierarchies in general with #161 , so I'm going to close this one.
